### PR TITLE
LibNameplateRegistry OM implementation

### PR DESCRIPTION
**Pull request by modhunter**
Using LibNameplateRegistry to dynamically add additional units (via nameplate[1-20]) to the OM tables

### CODE REVIEW

**Comment by modhunter** [comment]
I couldn't figure out how to load it properly other than:

`LibStub("AceAddon-3.0"):NewAddon("NerdPack", "LibNameplateRegistry-1.0");`

**Comment by modhunter** [comment]
I tried to get this to load via the following two approaches with no success:

```
local lnr = {};
LibStub("LibNameplateRegistry-1.0"):Embed(lnr);
```

&

```
local lnr = LibStub("LibNameplateRegistry-1.0")
```

If there isn't a better way to invoke the callback registration from the new lib, can we consider embedding AceAddon-3.0 to NeP?


An alternative approach would be to examine the innards of LibNameplateRegistry and re-implement however it's already collecting the unit info.  I personally don't like this approach as it's duplicated functionality that we need to add into NeP

**Comment by modhunter** [comment]
@MrTheSoulz Please hold off from merging this for now - exploring using this exclusivley instead of the target/mouseover/scanning party/raid targets

**Comment by modhunter** [comment]
For enemies, the nameplate thing works great.  However for friendly units (for healers), you need to turn on friendly nameplates which doesn't look very good.  So I think it's important to have the old-style party/raid members in the OM as well.

@MrTheSoulz I think this is good-to-go now

File: facerool.lua [hunk 1]
```diff
@@ -124,9 +125,45 @@ function NeP.Engine.FaceRoll()
...
+		self:LNR_UnregisterAllCallbacks();
+	end
+
+	function lnr:LNR_ON_NEW_PLATE(eventname, plateFrame, plateData)
+		--local R = "|cFFFF0000";
+		--local G = "|cFF00FF00";
+		--local W = "|r";
+		--print("LNR: (", plateData.unitToken, ") ", plateData.name, "'s nameplate appeared!");
+		--print("LNR: ", "It's a", R, plateData.type, W, "and", R, plateData.reaction, W, plateData.GUID and ("we know its GUID: " .. plateData.GUID) or "GUID not yet known");
+		nameplates[#nameplates+1] = plateData.unitToken
```
Comment (modhunter) [inline facerool.lua]: Is this a bad idea?  Will it keep growing the nameplates table indefinitely?

**Comment by MrTheSoulz** [comment]
LNR_Private:NAME_PLATE_UNIT_REMOVED

This should fix the table growing no?

**Comment by modhunter** [comment]
Yeah!  I think that's a better approach instead of using an integer as the key.  I'll implement this and test and report back here.

**Comment by modhunter** [comment]
@MrTheSoulz ok the changes to the nameplates handling is implemented and works great.  Can you please review?

I saw you mention something about disabling this while advanced OM is enabled.  We can do that and I added what I think would be the simplest way to do it.

**Pull request merged**